packages/theme-check-common/src/checks/unique-static-block-id/index.ts

```diff
@@ -0,0 +1,64 @@
+import { LiquidCheckDefinition, Severity, SourceCodeType } from '../../types';
+
+export const UniqueStaticBlockId: LiquidCheckDefinition = {
+  meta: {
+    code: 'UniqueStaticBlockId',
+    name: 'Prevent use of duplicated ids for blocks rendered statically',
+    docs: {
+      description:
+        'This check is aimed at preventing the use of duplicated ids for blocks rendered statically.',
+      url: 'https://shopify.dev/docs/themes/tools/theme-check/checks',
+      recommended: true,
+    },
+    type: SourceCodeType.LiquidHtml,
+    severity: Severity.ERROR,
+    schema: {},
+    targets: [],
+  },
+
+  create(context) {
+    const usedIds: string[] = [];
```

Comment:
A `Set` here should be more ideal for your usecase. Sets are faster than arrays for searching.